source/opt/eliminate_dead_input_components_pass.cpp

```diff
@@ -138,18 +155,12 @@ void EliminateDeadInputComponentsPass::ChangeArrayLength(Instruction& arr_var,
   analysis::Array new_arr_ty(arr_ty->element_type(),
                              arr_ty->GetConstantLengthInfo(length_id, length));
   analysis::Type* reg_new_arr_ty = type_mgr->GetRegisteredType(&new_arr_ty);
-  analysis::Pointer new_ptr_ty(reg_new_arr_ty, SpvStorageClassInput);
+  auto sclass = output_instead_ ? SpvStorageClassOutput : SpvStorageClassInput;
```

Comment:
Couldn't the storage class be taken from the source array? should remain the same no?

Reply:
OK.